src/Sylius/Bundle/ApiBundle/ApiPlatform/ConfigMergeManager.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\ApiPlatform;
+
+/** @experimental */
+class ConfigMergeManager
+{
+    public function mergeConfigs(...$configs): array
+    {
+        $resultingConfig = [];
+
+        foreach ($configs as $config) {
+            foreach ($config as $newKey => $newValue) {
+                $unsetNewKey = false;
+                if (is_string($newKey) && 1 === preg_match('/^(.*[^ ]) +\\(unset\\)$/', $newKey, $matches)) {
+                    [, $newKey] = $matches;
+                    $unsetNewKey = true;
+                }
+
+                if ($unsetNewKey) {
+                    unset($resultingConfig[$newKey]);
+
+                    if (null === $newValue) {
+                        continue;
+                    }
+                }
+
+                if (is_integer($newKey)) {
+                    $resultingConfig[] = $newValue;
+                } elseif (isset($resultingConfig[$newKey]) && is_array($resultingConfig[$newKey]) && is_array($newValue)) {
+                    $resultingConfig[$newKey] = $this->mergeConfigs($resultingConfig[$newKey], $newValue);
+                } else {
+                    $resultingConfig[$newKey] = $newValue;
+                }
```

Comment:
```suggestion
                if (is_integer($newKey)) {
                    $resultingConfig[] = $newValue;
                    
                    continue;
                } 
                
                if (isset($resultingConfig[$newKey]) && is_array($resultingConfig[$newKey]) && is_array($newValue)) {
                    $resultingConfig[$newKey] = $this->mergeConfigs($resultingConfig[$newKey], $newValue);
                    
                    continue;
                } 
                
                $resultingConfig[$newKey] = $newValue;
```
Perhaps this part could be extracted to the private method. But at least I would recommend simplification of the sctructure